### PR TITLE
Add redis opencensus metrics to the server

### LINKED ISSUE #266: Open Census for Redis
### TL;DR

Implement https://opencensus.io/integrations/redis/go/

### PR DESCRIPTION
**Pull request by taddari**
Fixes #266

## Proposed Changes

* Change redigo to opencensus/redigo
* Change redisstore to instrumented redisstore

**Release Note**

```release-note
Add redis opencensus metrics to the verification server, this will enable to monitor problems with redis connections itself.
```

### CODE REVIEW

**Comment by sethvargo** [comment]
Redis bits look good. Gonna defer to @icco on the metrics.

File: pkg/ratelimit/config.go [hunk 1]
```diff
@@ -71,9 +71,8 @@ func RateLimiterFor(ctx context.Context, c *Config) (limiter.Store, error) {
 		}
 
 		return redisstore.NewWithPool(config, &redis.Pool{
-			DialContext: func(ctx context.Context) (redis.Conn, error) {
-				return redis.DialContext(ctx, "tcp", addr,
-					redis.DialUsername(c.RedisUsername),
+			Dial: func() (redis.Conn, error) {
+				return redis.Dial("tcp", addr,
```
Comment (icco) [inline pkg/ratelimit/config.go:75]: Please continue to use DialWithContext https://godoc.org/github.com/opencensus-integrations/redigo/redis#DialWithContext

File: pkg/ratelimit/config.go [hunk 2]
```diff
@@ -71,9 +71,8 @@ func RateLimiterFor(ctx context.Context, c *Config) (limiter.Store, error) {
...
 					redis.DialPassword(c.RedisPassword),
```
Comment (icco) [inline pkg/ratelimit/config.go:76]: Setup with trace options: https://godoc.org/github.com/opencensus-integrations/redigo/redis#DialTraceOptions

File: pkg/ratelimit/config.go [hunk 3]
```diff
@@ -71,9 +71,8 @@ func RateLimiterFor(ctx context.Context, c *Config) (limiter.Store, error) {
 		}
 
 		return redisstore.NewWithPool(config, &redis.Pool{
-			DialContext: func(ctx context.Context) (redis.Conn, error) {
-				return redis.DialContext(ctx, "tcp", addr,
-					redis.DialUsername(c.RedisUsername),
```
Comment (icco) [inline pkg/ratelimit/config.go:76]: Hmm, I see open census doesn't have a dialusername, is that gonna be a problem? Should we patch opencensus to use that?

**Comment by icco** [comment]
> @taddari: The following test **failed**, say `/retest` to rerun all failed tests:
> 
> Test name	Commit	Details	Rerun command
> pull-en-server-release-unit	[1d545e6](https://github.com/google/exposure-notifications-verification-server/commit/1d545e6cd905686b7b531d4c730167ef90383d3b)	[link](https://oss-prow.knative.dev/view/gs/oss-prow/pr-logs/pull/google_exposure-notifications-verification-server/361/pull-en-server-release-unit/1298308110590414848)	`/test pull-en-server-release-unit`

This is just saying you need to run `go mod tidy`

File: pkg/ratelimit/config.go (hunk 3 above)
Comment (sethvargo) [reply pkg/ratelimit/config.go:76]: Won't matter for us. Redis 6+ allows for user/pass, but we're on 5

Comment (taddari) [reply pkg/ratelimit/config.go:76]: Yup, moreover the managed redis we provide on GCP actually don't support pass either.

File: pkg/ratelimit/limitware/middleware.go [hunk 4]
```diff
@@ -14,6 +14,7 @@ import (
 	"github.com/google/exposure-notifications-verification-server/pkg/controller"
 	"github.com/google/exposure-notifications-verification-server/pkg/database"
 	"github.com/google/exposure-notifications-verification-server/pkg/observability"
+	"github.com/opencensus-integrations/redigo/redis"
```
Comment (icco) [inline pkg/ratelimit/limitware/middleware.go:17]: nit: this should be in the last package group of imports

File: pkg/ratelimit/limitware/middleware.go [hunk 5]
```diff
@@ -110,6 +111,10 @@ func NewMiddleware(ctx context.Context, s limiter.Store, f httplimit.KeyFunc, op
 		return nil, fmt.Errorf("stat view registration failure: %w", err)
 	}
 
+	if err := view.Register(redis.ObservabilityMetricViews...); err != nil {
```
Comment (icco) [inline pkg/ratelimit/limitware/middleware.go:114]: It shouldn't be an issue, but we may in the future want to move this into the main functions of the servers, but for now this is fine here.

Comment (taddari) [reply pkg/ratelimit/limitware/middleware.go:114]: Noted.

File: pkg/ratelimit/limitware/middleware.go (hunk 4 above)
Comment (taddari) [reply pkg/ratelimit/limitware/middleware.go]: Done

**Pull request merged**